internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -954,6 +962,26 @@ func buildConnection(connection *egv1a1.ClientConnection) (*ir.ClientConnection,
 			irConnectionLimit.CloseDelay = ir.MetaV1DurationPtr(d)
 		}
 
+		if connection.ConnectionLimit.MaxConnectionDuration != nil {
```

Comment:
need this in multiple places - `if connection != nil && connection.ConnectionLimit != nil && ...`